src/client_side.cc

```diff
@@ -1265,7 +1277,8 @@ ConnStateData::parseHttpRequest(const Http1::RequestParserPointer &hp)
         // sync the buffers after parsing.
         inBuf = hp->remaining();
 
-        if (hp->needsMoreData()) {
+        incompleteHttpRequest_ = hp->needsMoreData();
```

Comment:
If possible, let's avoid introducing this new state (that would be easy to misinterpret in code that, for example, cannot tell whether the transaction has actually failed for another reason). If needed, this condition can be computed from scratch, similar to how Http::One::Server::parseOneRequest() computes a related condition, right?

In addition to the above concern, avoiding storing this condition will eliminate the need to explain why incompleteRequest() does not simply return incompleteHttpRequest_, implying that there is more to that "incomplete request" condition, and increasing raw incompleteHttpRequest_ misinterpretation risks.